File: 3-palindrome.py
k = int(input())
s = ""
if k==1:
	print('a')
elif k==2:
	print('aa')
elif k==3:
	print('bba')
else:
	for i in range(k/4):
		s+="aabb"
	h = k%4
	if h==0:
		pass
	elif h==1:
		s+='a'
	elif h==2:
		s+='aa'
	elif h==3:
		s+='aab'
print(s)

File: 3-palindrome.c
#include<stdio.h>
int main()
{
	int k,i;
	scanf("%d",&k);
	for(i=0;i<k/4;i++)
	{
		printf("aabb");
	}
	k=k%4;
	if(k==1)
		printf("a");
	else if(k==2)
		printf("aa");
	else if (k==3)
		printf("aab");
}